src/Umbraco.Core/Collections/ObservableDictionary.cs

```diff
@@ -235,15 +236,77 @@ public bool TryGetValue(TKey key, out TValue val)
 
         #endregion
 
+        /// <summary>
+        /// The exception that is thrown when a duplicate key inserted.
+        /// </summary>
+        /// <seealso cref="System.Collections.ObjectModel.ObservableCollection{TValue}" />
+        /// <seealso cref="System.Collections.Generic.IReadOnlyDictionary{TKey, TValue}" />
+        /// <seealso cref="System.Collections.Generic.IDictionary{TKey, TValue}" />
+        [Obsolete("Throw an ArgumentException when trying to add a duplicate key instead.")]
+        [Serializable]
         internal class DuplicateKeyException : Exception
```

Comment:
As this exception is internal and now unused, I think we could remove it completely...